app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,36 @@
+class FavoritesController < ApplicationController
+  def index
+    current_profile = Profile.find(params[:favorite_profile_id])
+    @collections = current_profile.favorites
+  end
+
+  def create
+    profile          = Profile.find(params[:favorite_profile_id])
+    favorite_profile = Profile.find_by(user_id: current_user.id)
```

Comment:
Did you mixed up this variables? variable `profile` taken from the params `favorite_profile_id`, but variable `favorite_profile` is your current user profile

Reply:
Yep I'll fix it